### ISSUE TITLE
Add support for Gemini

### ISSUE THREAD

**Issue by weilandia** [traceloop/openllmetry-ruby#7]
Here's a jenky half baked solution we're using in the context of our own `Llm::Chat::Gemini::ChatResponse`:

```ruby
     def response_for_log
        {
          "model" => model,
          "choices" => [
            {
              "message" => {
                "role" => "model",
                "content" => response.candidates.pluck("content").pluck("parts").flatten.pluck("text").join(" ")
              }
            }
          ],
          "usage" => {
            "total_tokens" => input_tokens + output_tokens,
            "prompt_tokens" => input_tokens,
            "completion_tokens" => output_tokens
          }
        }
      end

      def log_response
        require "traceloop/sdk"

        traceloop = Traceloop::SDK::Traceloop.new
        tracer = traceloop.instance_variable_get(:@tracer)

        tracer.in_span(model) do |span|
          t = traceloop.class::Tracer.new(span)
          t.log_prompt(model, prompt)
          t.log_response(response_for_log)
        end
      end
```

**Comment by weilandia** [comment]
Further, here's some of the weirdness we're considering to get to the interface we want:

```ruby
# frozen_string_literal: true

require "traceloop/sdk"

class TraceloopClient
  include Singleton

  attr_accessor :conn, :spanner

  def initialize
    self.conn = Traceloop::SDK::Traceloop.new
    self.spanner = conn.instance_variable_get(:@tracer)
  end

  def self.trace(name)
    instance.spanner.in_span(name) do |span|
      tracer = Traceloop::SDK::Traceloop::Tracer.new(span)
      yield(tracer)
    end
  end
end
```

**Comment by nirga** [comment]
Fixed with #8
